lightning/src/onion_message/packet.rs

```diff
@@ -239,15 +247,57 @@ impl Readable for ControlTlvs {
 	}
 }
 
-/// Reads padding to the end, ignoring what's read.
-pub(crate) struct Padding {}
+pub(crate) struct Padding {
+	padding_length: u16,
+}
+
+/// Reads padding to the end, and validates that the read bytes' content are 0s.
 impl Readable for Padding {
 	#[inline]
 	fn read<R: Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let mut padding_length = 0;
 		loop {
 			let mut buf = [0; 8192];
-			if reader.read(&mut buf[..])? == 0 { break; }
+			let read_bytes_len = reader.read(&mut buf[..])?;
+			if read_bytes_len == 0 { break; }
+			for n in 0..read_bytes_len {
+				if buf[n] != 0 as u8 { return Err(DecodeError::InvalidValue); }
```

Comment:
I don't think the padding is required to be 0s, or at least I can't find it in the spec ?

Reply:
Oh yeah, my bad! Good catch :)